### ISSUE TITLE
Phone hardware testing

### ISSUE THREAD

**Issue by Zulban** [osmosacademy/vr-campaign#22]
We need a person on our team to test our game releases on phone hardware regularly. They should have a laptop and a smartphone, either android or iphone. And probably a USB to phone cord too. It would be very helpful to know at all times what kind of frame rate our various test scenes are getting. The game must run well on a phone!

Peter or I can help set you up with "git" to get the latest versions of the game, and putting the game on your phone.

**Comment by peterwilkinson** [comment]
I've tested quite a few scenes, except for one of the latest demo builds. So far, so good.

**Comment by peterwilkinson** [comment]
The demo build has framerate issues, and is fairly constant (mostly 4fps on the Nexus 4...had up to 13fps but just briefly when shifting positions). Could be the amount of fish. The texture/lighting build took a bit of time when I opened the scene on the computer, and building with the assets took a little while as well. On a side note, the scene will need to move with the phone's sensors.

**Comment by Zulban** [comment]
Yep, it's a combination of the amount of fish and the huge and numerous textures. You can't have schools of 200 on the phone, only 50 or 100. Pretty sure the giant Unity terrain is no good too. The scene is not built for phones, it's to show what scripts I have working now. I really should have made that clear :P

**Comment by SamirSangani** [comment]
I dont have much experience with phones except for a couple of projects. They main things I found during that time were that old phones can handle ~ 30 - 50 K in polygons based on how much is rendered at a time. The biggest source of lag in my case was physics and transparent materials. I did not know this but transparent materials actually can solely bring the fps down to a halt. I understand that you would like to use billboards for the models, however if someone can make actual low poly models it would be a good comparison to see how they both do on fps. It would be better also to not have real time lighting/shadows to being with and then slowly introduce it in the future when we have achieved the target fps (you can try the scene with just the ambient light or use a good skybox).

**Comment by Zulban** [comment]
Good to know on poly counts. We are starting to collect some good information on performance.

I will say - we still need people to regularly test scenes on phone hardware. I can say that the phone can handle transparent billboard fishies quite well - as I recall from very early testing with Peter. But future effects/load is always going to be a mystery.

One of the reasons assembling a scene asap is so important is to get a handle on these limitations very early before we develop/find too many assets.

**Comment by SamirSangani** [comment]
Let me see if I can setup the cloud build tomorrow or Thursday night. This way I will be able to send across the links to everyone and they can try the builds on their phones without having to go through Unity as soon as a new scene is released.

**Comment by Zulban** [comment]
That sounds fabulous, thanks.
